[DOCS] Terminology experiment (#62592)
elastic/elasticsearch: docs/reference/modules/node.asciidoc

===============================================

[[master-node]]

> ==== Manager node

Reviewer: This section should really be about control nodes, although the text today is actually rather focussed on the manager responsibilities.

The manager node is responsible for lightweight cluster-wide actions such as
creating or deleting an index, tracking which nodes are part of the cluster,
and deciding which shards to allocate to which nodes. It is important for
cluster health to have a stable manager node.

Any control node that is not a <<voting-only-node,voting-only node>> can
become the manager through the <<modules-discovery, election process>>.

> IMPORTANT: Manager nodes must have access to the `data/` directory (just like

Reviewer: ```suggestion
IMPORTANT: Control nodes must have access to the `data/` directory (just like
```

Author: > I would prefer to use consistent terminology for the election process where it's necessary to mention it (which isn't often) but we could definitely consider changing that to something other than "election". In particular I like "selection", that fits the manager metaphor better IMO without completely losing the idea of votes and quorums.

I agree, there really isn't a need to explicitly call it an election.

`data` nodes) as this is where the cluster state is persisted between node
restarts.

[[dedicated-control-node]]
===== Dedicated control nodes

> It is important for the health of the cluster that the manager node has

Reviewer: I think we can generalise this:

```suggestion
It is important for the health of the cluster that the control nodes have
```

the resources it needs to fulfill its responsibilities. If the manager
is overloaded with other tasks then the cluster may not operate well. In
particular, indexing and searching your data can be very resource-intensive, so
in large or high-throughput clusters it is a good idea to avoid using the
control nodes for tasks such as indexing and searching. You can do this
by configuring three of your nodes to be dedicated control nodes.
Dedicated control nodes only have the `control` role, allowing them to

> focus on cluster management tasks. While control nodes can also behave as

Reviewer 2: ```suggestion
focus on controlling cluster management tasks. While control nodes can also behave as
```

I noticed this PR removed the verbiage around `manage` and `control. I wonder if this is because these verbs lack a clear delineation  

A `manager` node manages the cluster
A `control` node controls the cluster 

While the nouns `manager` and `control` have a much clearer delineation.

What does everyone think?

Author: I think we do need to provide a clearer description of what tasks the manager and control nodes perform. I was trying to avoid statements like "A manager node manages the cluster" that really don't provide any additional info.

This was just a quick pass to "try on" the terminology. There's definitely some fine-tuning to be done.

from clients to data nodes, it is better _not_ to use dedicated control nodes for
this purpose.

To create a dedicated control node, set:

[source,yaml]
-------------------
node.roles: [ control ]
-------------------

[[voting-only-node]]
===== Voting-only control nodes

A voting-only control node participates in the
<<modules-discovery,election process>> and cluster state publishing, 
but can never become the manager node. 
In particular, a voting-only node can serve as a tiebreaker
in elections.

To configure a control node as a voting-only node, include `control` and
`voting_only` in its list of roles. For example to create a voting-only data
node:

[source,yaml]
-------------------
node.roles: [ data, control, voting_only ]
-------------------

IMPORTANT: The `voting_only` role requires the {default-dist} of {es} and is not
supported in the {oss-dist}. If you use the {oss-dist} and add the `voting_only`
role then the node will fail to start.  Also note that only nodes with the
`control` role can have the `voting_only` role.

High availability (HA) clusters require at least three control nodes, at
least two of which are not voting-only nodes. This enables a cluster to

> promote a control node to manager even if one of the control nodes fails.

Reviewer: ```suggestion
elect a control node to manager even if the current manager node fails.
```

Author: Was sort of "trying this on" as an alternative to elect. I like the idea of select/selection.
